### ISSUE TITLE
Scarcs: numerical instability with single process

### ISSUE THREAD

**Issue by sbenkorichi** [firemodels/fds#6546]
Susan,
When you have time, could you please check how you can fix this. The cases of scarcs run only with mpi process at the moment, Running it in single process ends up in numerical instability.
Thanks!

**Comment by rmcdermo** [comment]
@SusanKilian Salah noticed while testing the bundle that ScaRC the dancing_eddies_scarc.fds case gives a numerical instability if run with only a single process.  We are trying to get a bundle out and I would really like ScaRC to be functional for it, even though it is not advertised.  This might be the last minor release before we switch over to v7.  Thanks.

**Comment by SusanKilian** [comment]
I’ll have a look at it right now. In fact, I have changed the way boundary conditions are handled. I suspect that I have made a mistake with it.

**Comment by rmcdermo** [comment]
Try compiling with Gnu.

**Comment by sbenkorichi** [comment]
Randy,
I'm not sure if Gnu would help for this, because with Gnu it passes. I found it fails when running it with Intel. Not sure why.

**Comment by SusanKilian** [comment]
Ok, I could reproduce it now and am working on it …

**Comment by rmcdermo** [comment]
It is choking on this line in dump:
```
IF (ICYC>0) THEN
   WRITE(LU_OUTPUT,'(//A,F12.3)') ' Time Stepping Wall Clock Time (s): ',SUM(T_USED(1:N_TIMERS))
   WRITE(LU_OUTPUT,'(  A,F12.3)') ' Total Elapsed Wall Clock Time (s): ',CURRENT_TIME()-WALL_CLOCK_START
ENDIF
```
I know Marcos recently added timings for pressure solver stuff.

**Comment by rmcdermo** [comment]
Well, the previous problem had nothing to do with the instability.  But it was a problem nonetheless.

**Comment by SusanKilian** [comment]
Okay, I see … I'm on the track of the single-process problem

**Comment by marcosvanella** [comment]
Susan, you probably noted that UscaRC is behaving the same way with one process-multiple meshes.

**Comment by SusanKilian** [comment]
Thanks for the hint. Yes, that’s true. 

I have already discovered that the error is related to the matrix-vector multiplication which is handled differently by the multi-process and single-process versions. And this applies equally to ScaRC and UScaRC. The single-process version used to run once, but in the meantime I've changed a lot and I guess that I didn’t check it for a long time. Anyway, I am confident that I will find the mistake soon.
